external/fv3fit/fv3fit/_shared/jacobian.py

```diff
@@ -73,3 +72,56 @@ def compute_standardized_jacobians(model, data, input_variables):
     jacobians = get_jacobians(model, avg_profiles)
     std_jacobians = standardize_jacobians(jacobians, data)
     return std_jacobians
+
+
+def standardize_jacobians_by_vertical_level(
+    all_jacobians: Mapping[str, OutputSensitivity],
+    sample: Mapping[str, tf.Tensor],
+    units: str = "dimensionless",
+) -> Mapping[str, OutputSensitivity]:
+    """
+    Generate sensitivity jacobians for each output of a model and
+    standardize (dimensionless or output units) for easy
+    inter-variable comparison.
+
+    The scaling uses the standard deviation of each vertical level
+    for the input (std_input) and output
+    (std_output) sample. The associated
+    jacobian result is multiplied by [ std_input / std_output ].
+    If units are "output" the associated jacobian result is multiplied
+    by [ std_input ].
```

Comment:
I worry the code in this module is being too tightly coupled to the needs of downstream plots. I think the APIs should be refactored.

As with `standardize_jacobians`, this function does two things...computes the std and scales the jacobian. I suggest a refactor below.